<launch>
    <!-- Map server -->
    <node name="map_view" pkg="map_server" type="map_server" args="/home/breach/catkin_ws/maps/map_actual.yaml" >
        <param name="frame_id" value="map" />
    </node>

    <!-- map TF -->
<!--
    <node name="map_publisher" pkg="robot_setup_tf" type="map_publisher" />
-->

    <!-- AMCL -->
    <node pkg="amcl" type="amcl" name="amcl" output="screen">
        <remap from="scan" to="scan" />
        <!-- Publish scans from best pose at a max of 10 Hz -->

        <param name="initial_pose_x" value="14.297"/>
        <param name="initial_pose_y" value="20.368"/>
            <param name="initial_pose_a" value="-1.511"/>

        <param name="odom_model_type" value="diff"/>
        <param name="odom_alpha1" value="0.2" />
        <param name="odom_alpha2" value="0.1"/>
        <param name="odom_alpha3" value="0.02"/>
        <param name="odom_alpha4" value="0.05"/>

        <param name="odom_frame_id" value="odom"/>
        <param name="base_frame_id" type="str" value="base_link" />
        <param name="global_frame_id" type="str" value="map" />

        <param name="gui_publish_rate" value="2.0"/>
        <param name="laser_max_beams" value="50"/>

        <param name="min_particles" value="100"/>
        <param name="max_particles" value="1000"/>

        <param name="kld_err" value="0.01"/>
        <param name="kld_z" value="0.99"/>
        <param name="transform_tolerance" value="0.1" />

        <param name="update_min_d" value="0.5"/>
        <param name="update_min_a" value="0.05"/>

        <param name="resample_interval" value="3"/>

        <param name="laser_z_hit" value="0.5"/>
        <param name="laser_z_short" value="0.05"/>
        <param name="laser_z_max" value="0.05"/>
        <param name="laser_z_rand" value="0.5"/>
        <param name="laser_sigma_hit" value="0.2"/>
        <param name="laser_lambda_short" value="0.1"/>
        <param name="laser_model_type" value="likelihood_field_prob"/>
        <param name="laser_likelihood_max_dist" value="2.0"/>

        <param name="recovery_alpha_slow" value="0.0"/>
        <param name="recovery_alpha_fast" value="0.0"/>

        <param name="use_map_topic" value="true" />
        <param name="first_map_only" value="true" />
    </node>

    <!-- MOVE BASE -->
    <node pkg="move_base" type="move_base" name="move_base" output="screen">
        <param name="controller_frequency" value="5.0" />
        <param name="controller_patience" value="30.0" />
        <param name="planner_frequency" value="0.1" />
        <param name="planner_patience" value="30.0" />
        <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />)
        <param name="clearing_rotation_allowed" value="false" />
        <param name="max_planning_retries" value="100" />
        <param name="recovery_behavior_enabled" value="false" />

        <rosparam file="$(find navigation)/config/local_costmap_params.yaml" command="load" />
        <rosparam file="$(find navigation)/config/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find navigation)/config/teb_local_planner.yaml" command="load" />
    </node>

    <node name="recovery" pkg="breach" type="breach_recovery.py" output="screen"/>

</launch>
